documentation/library_guide/introduction.rst

```diff
@@ -166,7 +166,7 @@ Known Limitations
 * For ``max_element``, ``min_element``, ``minmax_element``, ``partial_sort``, ``partial_sort_copy``, ``sort``, ``stable_sort``
   the dereferenced value type of the provided iterators should satisfy the ``DefaultConstructible`` requirements.
 * For ``remove``, ``remove_if``, ``unique`` the dereferenced value type of the provided
-  iterators should be ``MoveConstructible``.
+  iterators should be ``CopyConstructible``.
```

Comment:
Could you please point out to the place in the code where copy construction is needed?

Reply:
I've asked Mikhail to note here: https://github.com/uxlfoundation/oneDPL/pull/2100#discussion_r2060790704 with no much analysis. So, let me assist.

This pattern attracted my attention:
https://github.com/uxlfoundation/oneDPL/blob/e6fa213d7f2688d4d6d6e84c71a00b3aecfe4d1b/include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h#L991-L1012

That's because it uses `__brick_copy`. In fact, it requires `CopyAssignable`.

Reply:
> Could you please point out to the place(s) in the code where copy construction is needed?

remove, remove_if, unique implementations are based on temporary buffer. So, an implementation must "copy-constsruct" each element or "default construct + assign each element" for buffer initialization. After an implementation must copy/assign each element back.